### DISCUSSION TITLE
如何动态修改graph的配置?

### DISCUSSION THREAD

**Discussion by sheila1227** [antvis/X6#2191] [Q&A] [answered]
如下图，我准备了一个radioButtonGroup用来控制画出来的线是什么颜色（选success画出绿色连接线，选failure画出红色连接线等），操作顺序是先选择箭头类型，再在画布上操作。于是我在new Graph的option内传入了connecting的createEdge方法，方法体内取state上的arrowType来决定画线颜色。
但是每次在radioButtonGroup选择后，state上的值都变化了，但当createEdge被调用时，这个createEdgeInConnecting内拿到的arrowType值永远是最开始的值，没有变过。
请问有什么办法能动态修改graph.option里的createEdge吗？
<img width="570" alt="image" src="https://user-images.githubusercontent.com/9313197/172527191-673013a0-8dca-4925-8ec3-ce02304be6d7.png">
相关代码：
![image](https://user-images.githubusercontent.com/9313197/172528029-0fc86ef4-8239-4639-ba7e-e7596c63eacd.png)
![image](https://user-images.githubusercontent.com/9313197/172528043-afc14399-a0a9-4606-89d7-f2b9c7dcd587.png)

**Comment by NewByVector** [comment, marked as answer]
可以这样：
```ts
graph.options.connecting.createEdge = () => {
  return graph.createEdge({
    attrs: {
      line: {
        stroke: 'blue'
      }
    }
  })
}
```
